Crash on user authentication

I'm experiencing a crash when I'm successfully authenticated and try to send.

```
# smtpd -dv
debug: init ssl-tree
info: loading pki information for foo.bar
debug: init ca-tree
debug: init ssl-tree
info: loading pki keys for foo.bar
debug: using "fs" queue backend
debug: using "ramqueue" scheduler backend
debug: using "ram" stat backend
info: OpenSMTPD 6.0.3-portable starting
debug: init ssl-tree
info: loading pki information for foo.bar
debug: init ca-tree
debug: init ssl-tree
info: loading pki keys for foo.bar
debug: using "fs" queue backend
debug: using "ramqueue" scheduler backend
debug: using "ram" stat backend
setup_peer: control -> klondike[12376] fd=5
setup_peer: control -> lookup[12378] fd=6
setup_peer: control -> pony express[12379] fd=7
setup_peer: control -> queue[12380] fd=8
setup_peer: control -> scheduler[12381] fd=9
debug: init ssl-tree
info: loading pki information for foo.bar
debug: init ca-tree
debug: init ssl-tree
info: loading pki keys for foo.bar
debug: using "fs" queue backend
debug: using "ramqueue" scheduler backend
debug: using "ram" stat backend
setup_peer: klondike -> control[12377] fd=5
setup_peer: klondike -> pony express[12379] fd=6
setup_done: ca[12376] done
setup_done: control[12377] done
setup_proc: control done
setup_proc: klondike done
debug: init ssl-tree
info: loading pki information for foo.bar
debug: init ca-tree
debug: init ssl-tree
info: loading pki keys for foo.bar
debug: using "fs" queue backend
debug: using "ramqueue" scheduler backend
debug: using "ram" stat backend
setup_peer: pony express -> control[12377] fd=5
setup_peer: pony express -> klondike[12376] fd=6
setup_peer: pony express -> lookup[12378] fd=7
setup_peer: pony express -> queue[12380] fd=8
debug: init ssl-tree
info: loading pki information for foo.bar
debug: init ca-tree
debug: init ssl-tree
info: loading pki keys for foo.bar
debug: using "fs" queue backend
debug: init ssl-tree
info: loading pki information for foo.bar
debug: init ca-tree
debug: init ssl-tree
info: loading pki keys for foo.bar
debug: using "fs" queue backend
debug: using "ramqueue" scheduler backend
debug: using "ram" stat backend
setup_peer: queue -> control[12377] fd=5
setup_peer: queue -> pony express[12379] fd=6
debug: init ssl-tree
info: loading pki information for foo.bar
debug: init ca-tree
debug: init ssl-tree
info: loading pki keys for foo.bar
setup_peer: queue -> lookup[12378] fd=7
debug: using "fs" queue backend
debug: using "ramqueue" scheduler backend
debug: using "ram" stat backend
setup_peer: queue -> scheduler[12381] fd=8
setup_peer: lookup -> control[12377] fd=5
debug: using "ramqueue" scheduler backend
debug: using "ram" stat backend
setup_peer: lookup -> pony express[12379] fd=6
setup_peer: scheduler -> control[12377] fd=5
setup_peer: lookup -> queue[12380] fd=7
setup_peer: scheduler -> queue[12380] fd=6
setup_done: lka[12378] done
setup_proc: lookup done
setup_proc: pony express done
setup_done: pony[12379] done
setup_done: queue[12380] done
setup_proc: queue done
debug: ca_engine_init: using RSA privsep engine
setup_proc: scheduler done
setup_done: scheduler[12381] done
smtpd: setup done
debug: bounce warning after 4h
debug: parent_send_config_ruleset: reloading
debug: parent_send_config: configuring pony process
debug: parent_send_config: configuring ca process
debug: smtp: listen on 1.2.3.4 port 25 flags 0x401 pki "foo.bar" ca ""
debug: smtp: listen on 1.2.3.4 port 587 flags 0x469 pki "foo.bar" ca ""
debug: pony: rsae_init
debug: init private ssl-tree
debug: pony: rsae_init
debug: smtp: will accept at most 524283 clients
debug: queue: done loading queue into scheduler
debug: smtpd: scanning offline queue...
debug: smtpd: offline scanning done
debug: smtp: new client on listener: 0x562b984c5e20
3325930c6f234157 smtp event=connected address=XXX.XXX.XXX.XXX host=XXXXXXXXXXXXXXX
debug: lka: looking up pki "foo.bar"
debug: session_start_ssl: switching to SSL
debug: pony: rsae_priv_enc
3325930c6f234157 smtp event=starttls address=XXX.XXX.XXX.XXX host=XXXXXXXXXXXXXXX ciphers="version=TLSv1.2, cipher=ECDHE-RSA-AES256-GCM-SHA384, bits=256"
debug: lka: authenticating for creds:user@foo.bar
debug: control -> lookup: pipe closed
debug: control agent exiting
debug: parent -> lka: pipe closed
smtpd: process lka socket closed
debug: queue -> lookup: pipe closed
debug: queue agent exiting
debug: scheduler -> queue: pipe closed
debug: scheduler agent exiting
debug: ca -> parent: pipe closed
debug: ca agent exiting
debug: pony -> lookup: pipe closed
debug: pony agent exiting

```

from `./configure`
```
checking crypt.h usability... yes
checking crypt.h presence... yes
checking for crypt.h... yes
```

from `make`
```
gcc -DHAVE_CONFIG_H -I. -I..  -I../smtpd -I../openbsd-compat -I/usr/include   -g -O2  -fPIC -DPIC -Wall -Wpointer-arith -Wuninitialized -Wsign-compare -Wformat-security -Wsizeof-pointer-memaccess -Wno-pointer-sign -Wno-unused-result -fno-strict-aliasing -fno-builtin-memset -D_BSD_SOURCE -D_DEFAULT_SOURCE  -c -o crypt_checkpass.o crypt_checkpass.c
crypt_checkpass.c: In function 'crypt_checkpass':
crypt_checkpass.c:22:13: warning: implicit declaration of function 'crypt'; did you mean 'chroot'? [-Wimplicit-function-declaration]
  if (strcmp(crypt(pass, goodhash), goodhash) == 0)
             ^~~~~
             chroot
crypt_checkpass.c:22:13: warning: passing argument 1 of 'strcmp' makes pointer from integer without a cast [-Wint-conversion]
In file included from crypt_checkpass.c:9:0:
```

This looks like `crypt.h` gets detected but not included in `crypt_checkpass` leading to undefined behaviour (which apparently works out as long `--with-pie` is not used).

Setting `-DHAVE_CRYPT_H` in `CFLAGS` results in a working binary (with pie).

So the remaing step is to check, why `HAVE_CRYPT_H` is not defined in `crypt_checkpass`.

--- a/openbsd-compat/crypt_checkpass.c
+++ b/openbsd-compat/crypt_checkpass.c
@@ -1,5 +1,6 @@
 /* OPENBSD ORIGINAL: lib/libc/crypt/cryptutil.c */
 
+#include "includes.h"
 #include <errno.h>
 #ifdef HAVE_CRYPT_H
 #include <crypt.h>
@@ -10,6 +11,8 @@
 int
 crypt_checkpass(const char *pass, const char *goodhash)
 {
+	char *c;
+
 	if (goodhash == NULL)
 		goto fail;
 
@@ -17,7 +20,11 @@ crypt_checkpass(const char *pass, const char *goodhash)
 	if (strlen(goodhash) == 0 && strlen(pass) == 0)
 		return 0;
 
-	if (strcmp(crypt(pass, goodhash), goodhash) == 0)
+	c = crypt(pass, goodhash);
+	if (c == NULL)
+		goto fail;
+
+	if (strcmp(c, goodhash) == 0)
 		return 0;
 
 fail:

